api/client.go

```diff
@@ -470,6 +462,50 @@ func (c *Config) ReadEnvironment() error {
 	return nil
 }
 
+// ParseAddress transforms the provided address into a url.URL and handles
+// the case of Unix domain sockets by setting the DialContext in the
+// configuration's HttpClient.Transport.
```

Comment:
Can you add a note in this comment that this function should always be called with `c.modifyLock` held for write access?